libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/location/LocationViewModel.kt

```diff
@@ -22,24 +24,24 @@ sealed class LocationAction {
 class LocationViewModel : UIViewModel<Location?, LocationAction>(null) {
     val navigationLocationProvider = NavigationLocationProvider()
 
+    /**
+     * Accessor function to get the last [Point]
+     */
     val lastPoint: Point?
         get() = navigationLocationProvider.lastLocation?.run {
             Point.fromLngLat(longitude, latitude)
         }
 
-    private val locationObserver: LocationObserver = object : LocationObserver {
-        override fun onNewRawLocation(rawLocation: Location) {
-            // We are not sending LocationAction.UpdateRaw here, because we are expecting to
-            // receive locationMatcherResult
-        }
-
-        override fun onNewLocationMatcherResult(locationMatcherResult: LocationMatcherResult) {
-            navigationLocationProvider.changePosition(
-                location = locationMatcherResult.enhancedLocation,
-                keyPoints = locationMatcherResult.keyPoints,
-            )
-            invoke(LocationAction.Update(locationMatcherResult.enhancedLocation))
+    /**
+     * Suspend until a non-null location is available.
+     */
+    suspend fun firstLocation(): Location {
+        var nonNullLocation: Location? = navigationLocationProvider.lastLocation
```

Comment:
Can we have a better name for this variable?

Reply:
Do you have a suggestion?

Reply:
Why not simply `location`